libbeat/scripts/generate_template.py

```diff
@@ -51,95 +58,148 @@ def fields_to_es_template(input, output, index):
                 "_all": {
                     "norms": False
                 },
-                "properties": {},
-                "dynamic_templates": [{
-                    "template1": {
-                        "match_mapping_type": "string",
-                        "mapping": {
-                            "type": "keyword",
-                            "index": defaults["index"],
-                            "doc_values": defaults["doc_values"],
-                            "ignore_above": defaults["ignore_above"]
-                        }
-                    }
-                }]
+                "properties": {}
             }
         }
     }
 
     properties = {}
+    dynamic_templates = []
     for doc, section in docs.items():
-        if doc not in ["version", "defaults"]:
-            prop = fill_section_properties(section, defaults)
+        if doc != "defaults":
+            prop, dynamic = fill_section_properties(section, defaults, "")
             properties.update(prop)
+            dynamic_templates.extend(dynamic)
 
     template["mappings"]["_default_"]["properties"] = properties
+    if len(dynamic_templates) > 0:
+        template["mappings"]["_default_"]["dynamic_templates"] = \
+            dynamic_templates
 
     json.dump(template, output,
               indent=2, separators=(',', ': '),
               sort_keys=True)
 
 
-def fill_section_properties(section, defaults):
+def dedot(group):
+    """
+    Walk the tree and transform keys like "beat.name" and "beat.hostname" into
```

Comment:
It's needed for docs, I think, which have a pretty rigid structure. Maybe we could get rid of it, but I was afraid of breaking existing docs.